### PR TITLE
Import Config tests, remove last refs to `Chef::Stuff`

### PR DESCRIPTION
**Pull request by danielsdeleo**
* Import specs for Config
* Remove unfixed references to `Chef::Whatever`
* Pull in changes to `Chef::Config` that occurred after we started

When this is merged, I believe we'll be ready to subtree merge this repo into chef.

@chef/client-core

### CODE REVIEW

**Comment by jdmundrawala** [comment]
:+1:

File: spec/unit/config_spec.rb [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+
+        it "ChefConfig::Config[:ssl_verify_mode] defaults to :verify_peer" do
+          expect(ChefConfig::Config[:ssl_verify_mode]).to eq(:verify_peer)
+        end
+
+        it "ChefConfig::Config[:ssl_ca_path] defaults to nil" do
+          expect(ChefConfig::Config[:ssl_ca_path]).to be_nil
+        end
+
+        # TODO can this be removed?
```
Comment (tyler-ball) [inline spec/unit/config_spec.rb]: Is this TODO relevant?

**Comment by tyler-ball** [comment]
:+1:

File: spec/unit/config_spec.rb (hunk 1 above)
Comment (danielsdeleo) [inline spec/unit/config_spec.rb]: Heh, I only copied this from Chef and then fixed references to `Chef::Whatever`. But it certainly looks like we run this test with both windows and not windows 3 lines above, and then run it again on unix.

Comment (danielsdeleo) [inline spec/unit/config_spec.rb]: Derp, that's `ssl_ca_path` not `ssl_ca_file`.

**Pull request merged**